[Bug] [Master] Failed to pass global parameters between parent and sub process

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

When the parent process has global parameters, but the sub process does not have global parameters, a null pointer error will be triggered
`[ERROR] 2021-12-31 14:06:44.898 org.apache.dolphinscheduler.server.master.runner.MasterSchedulerService:[215] - scan command error
java.lang.UnsupportedOperationException: null
        at java.util.AbstractList.add(AbstractList.java:148)
        at java.util.AbstractList.add(AbstractList.java:108)
        at org.apache.dolphinscheduler.service.process.ProcessService.joinGlobalParams(ProcessService.java:1025)
        at org.apache.dolphinscheduler.service.process.ProcessService.setSubProcessParam(ProcessService.java:992)
        at org.apache.dolphinscheduler.service.process.ProcessService.handleCommand(ProcessService.java:226)
        at org.apache.dolphinscheduler.service.process.ProcessService$$FastClassBySpringCGLIB$$ed138739.invoke(<generated>)
`

### What you expected to happen

Global parameters are passed normally

### How to reproduce

Configure two processes, where the sub process has no global parameters, and the parent process  has global parameters

### Anything else

_No response_

### Version

2.0.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java
+++ b/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java
@@ -1064,6 +1064,7 @@ public class ProcessService {
         List<Property> subPropertyList = JSONUtils.toList(subGlobalParams, Property.class);
         subPropertyList = new ArrayList<>(subPropertyList);
 
+        
         Map<String, String> subMap = subPropertyList.stream().collect(Collectors.toMap(Property::getProp, Property::getValue));
 
         for (Property parent : parentPropertyList) {

--- a/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java
+++ b/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java
@@ -1063,7 +1063,6 @@ public class ProcessService {
         List<Property> parentPropertyList = JSONUtils.toList(parentGlobalParams, Property.class);
         List<Property> subPropertyList = JSONUtils.toList(subGlobalParams, Property.class);
         subPropertyList = new ArrayList<>(subPropertyList);
-
         
         Map<String, String> subMap = subPropertyList.stream().collect(Collectors.toMap(Property::getProp, Property::getValue));
 

--- a/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java
+++ b/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java
@@ -1062,6 +1062,7 @@ public class ProcessService {
 
         List<Property> parentPropertyList = JSONUtils.toList(parentGlobalParams, Property.class);
         List<Property> subPropertyList = JSONUtils.toList(subGlobalParams, Property.class);
+        subPropertyList = new ArrayList<>(subPropertyList);
 
         Map<String, String> subMap = subPropertyList.stream().collect(Collectors.toMap(Property::getProp, Property::getValue));
 

